src/file.ts

```diff
@@ -4168,7 +4167,17 @@ class File extends ServiceObject<File, FileMetadata> {
       highWaterMark: options?.highWaterMark,
       universeDomain: this.bucket.storage.universeDomain,
       [GCCL_GCS_CMD_KEY]: options[GCCL_GCS_CMD_KEY],
-    });
+    };
+
+    let uploadStream: resumableUpload.Upload;
+
+    try {
+      uploadStream = resumableUpload.upload(cfg);
+    } catch (error) {
+      dup.destroy(error as Error);
+      this.storage.retryOptions.autoRetry = false;
```

Comment:
There is no need to set this here.

Reply:
Seems this is a unrecoverable error.

https://github.com/googleapis/nodejs-storage/blob/18eef67892af6c7311a12d4c22579e240149eb02/src/resumable-upload.ts#L350-L354

Reply:
Retries for resumable uploads do not happen at this layer. Turning this to false here is shutting off retries for other operations as well.

Reply:
OK, NP